### ISSUE TITLE
Set alternate Windows report mode for tests and research minidumps

### ISSUE THREAD

**Issue by bentoi** [zeroc-ice/ice#72]
Right now, when running an exe in debug mode and an assert occurs, a dialog pops up with "Abort", "Retry" and "Cancel" buttons which ends up causing the tests to hang. We could modify this default behavior to instead just print a message on the console, see: https://msdn.microsoft.com/fr-fr/library/1y71x448.aspx

We should also check if Windows minidumps can help debug failures in crash or hang situations

http://www.debuginfo.com/articles/effminidumps.html
https://support.microsoft.com/en-us/help/241215/how-to-use-the-userdump.exe-tool-to-create-a-dump-file
https://support.microsoft.com/en-us/help/931673/how-to-create-a-user-mode-process-dump-file-in-windows-vista-and-in-wi

**Comment by pepone** [comment]
```
    _CrtSetReportMode(_CRT_ASSERT, _CRTDBG_MODE_FILE);
    _CrtSetReportFile(_CRT_ASSERT, _CRTDBG_FILE_STDERR);

    _CrtSetReportMode(_CRT_ERROR, _CRTDBG_MODE_FILE);
    _CrtSetReportFile(_CRT_ERROR, _CRTDBG_FILE_STDERR);

    _CrtSetReportMode(_CRT_WARN, _CRTDBG_MODE_FILE);
    _CrtSetReportFile(_CRT_WARN, _CRTDBG_FILE_STDERR);

    _set_abort_behavior(_CALL_REPORTFAULT, _CALL_REPORTFAULT);
```

Something like this will take care of the dialog and trigger the reports if [WER](https://msdn.microsoft.com/en-us/library/windows/desktop/bb513638(v=vs.85).aspx) is enabled

**Comment by pepone** [comment]
We should fix that along #62

**Comment by pepone** [comment]
That is now done with the TestHelper class ctor

https://github.com/zeroc-ice/ice/blob/7919c21dc8f936f7e99650f55b27dff33c4d1605/cpp/test/Common/TestHelper.cpp#L70-L80
